### ISSUE TITLE
生成器和判别器输入和输出数据维度不匹配

### ISSUE THREAD

**Issue by LLGISer** [Helomin/STMixGAN#2]
作者您好，在复现您提供的代码时，发现生成器和判别器的输入和输出数据出现维度不匹配问题，按照您提供代码里的逻辑，如在生成器中train_imgs为[8,5,1,128,128],而train_fake却为[8,5,128,128]，这导致将train_input和train_fake进行cat操作时会造成维度不匹配。我试图修改这一问题，但发现逻辑出现了紊乱，不能完全的复现原论文的所提供的方法。希望您可以解答，非常感谢！
            for (itr, train_imgs) in enumerate(self.train_loader):
                train_imgs = train_imgs.type(torch.cuda.FloatTensor)
                train_input = train_imgs[:, :5]
                train_target = train_imgs[:, 5:]

                # train D
                if (itr + 1) % 2 == 0:
                    train_fake = self.gen(train_input).detach()

                    self.optimizer_critic.zero_grad()

                    input_real = torch.cat([train_input, train_target], dim=1)
                    input_fake = torch.cat([train_input, train_fake], dim=1)
                    critic_real = self.critic(input_real)
                    critic_fake = self.critic(input_fake)

                    gp = gradient_penalty_div(
                        self.critic, input_real, input_fake)
                    critic_loss = - \
                        torch.mean(critic_real) + torch.mean(critic_fake) + gp
                    critic_loss.backward()
                    self.optimizer_critic.step()

**Comment by LLGISer** [comment]
您好，方便解答一下这个问题吗？非常感谢

**Comment by liviadeazevedo** [comment]
Hi!
I was able to run the project code, but I had to make some modifications. To fix this problem, I used torch.reshape() to adjust the incompatible tensor sizes. I encountered the same size issue in other parts of the code, but I applied the same solution. I'm not sure if this is the correct way to run the code, but with these changes, the training was able to run. Here is my code:

            
                # train D
                if (itr + 1) % 2 == 0:
                    train_fake = self.gen(train_input).detach()

                    # Added this line
                    train_fake = torch.reshape(train_fake, [train_fake.shape[0], 1, train_fake.shape[1], train_fake.shape[2],
                                                          train_fake.shape[3]])

                    self.optimizer_critic.zero_grad()

                    input_real = torch.cat([train_input, train_target], dim=1)
                    input_fake = torch.cat([train_input, train_fake], dim=1)

                    # Added these lines
                    input_real = torch.reshape(input_real,[input_real.shape[0], input_real.shape[1], input_real.shape[3], input_real.shape[4]])
                    input_fake = torch.reshape(input_fake,[input_fake.shape[0], input_fake.shape[1], input_fake.shape[3],input_fake.shape[4]])

                    # Added these lines
                    input_real = input_real[:, -2:, :, :]
                    input_fake = input_fake[:, -2:, :, :]

                    critic_real = self.critic(input_real)
                    critic_fake = self.critic(input_fake)

                    gp = gradient_penalty_div(
                        self.critic, input_real, input_fake)
                    critic_loss = - \
                        torch.mean(critic_real) + torch.mean(critic_fake) + gp
                    critic_loss.backward()
                    self.optimizer_critic.step()
